pkg/cache/cache_test.go

```
@@ -69,185 +125,180 @@ var _ = Describe("Informer Cache", func() {
+
+			By("verifying that the returned service looks reasonable")
+			Expect(svc.Name).To(Equal("kubernetes"))
+			Expect(svc.Namespace).To(Equal("default"))
+		})
+
+		It("should support filtering by labels", func() {
+			By("listing pods with a particular label")
+			// NB: each pod has a "test-label" equal to the pod name
+			out := kcorev1.PodList{}
+			Expect(informerCache.List(context.Background(), client.InNamespace(testNamespaceTwo).
+				MatchingLabels(map[string]string{"test-label": "test-pod-2"}), &out)).NotTo(HaveOccurred())
+
+			By("verifying the returned pods have the correct label")
+			Expect(out.Items).NotTo(BeEmpty())
+			Expect(out.Items).Should(HaveLen(1))
+			actual := out.Items[0]
+			Expect(actual.Labels["test-label"]).To(Equal("test-pod-2"))
+		})
+
+		It("should be able to list objects by namespace", func() {
+			By("listing pods in test-namespace-1")
+			listObj := &kcorev1.PodList{}
+			Expect(informerCache.List(context.Background(),
+				client.InNamespace(testNamespaceOne),
+				listObj)).NotTo(HaveOccurred())
+
+			By("verifying that the returned pods are in test-namespace-1")
+			Expect(listObj.Items).NotTo(BeEmpty())
+			Expect(listObj.Items).Should(HaveLen(1))
+			actual := listObj.Items[0]
+			Expect(actual.Namespace).To(Equal(testNamespaceOne))
+		})
+
+		It("should deep copy the object unless told otherwise", func() {
+			By("retrieving a specific pod from the cache")
+			out := &kcorev1.Pod{}
+			podKey := client.ObjectKey{Name: "test-pod-2", Namespace: testNamespaceTwo}
+			Expect(informerCache.Get(context.Background(), podKey, out)).NotTo(HaveOccurred())
+
+			By("verifying the retrieved pod is equal to a known pod")
+			Expect(out).To(Equal(knownPod2))
+
+			By("altering a field in the retrieved pod")
+			*out.Spec.ActiveDeadlineSeconds = 4
+
+			By("verifying the pods are no longer equal")
+			Expect(out).NotTo(Equal(knownPod2))
+		})
+
+		It("should error out for missing objects", func() {
```

nit: `should return an error if the object is not found`